lnd.go

```diff
@@ -666,46 +667,58 @@ func bakeMacaroon(ctx context.Context, svc *macaroons.Service,
 	return mac.M().MarshalBinary()
 }
 
-// genMacaroons generates three macaroon files; one admin-level, one for
-// invoice access and one read-only. These can also be used to generate more
-// granular macaroons.
-func genMacaroons(ctx context.Context, svc *macaroons.Service,
-	admFile, roFile, invoiceFile string) error {
+// saveMacaroon bakes a macaroon with the specified macaroon permissions and
+// writes it to a file with the given filename and file permissions.
+func saveMacaroon(ctx context.Context, svc *macaroons.Service, filename string,
+	macaroonPermissions []bakery.Op, filePermissions fs.FileMode) error {
 
-	// First, we'll generate a macaroon that only allows the caller to
-	// access invoice related calls. This is useful for merchants and other
-	// services to allow an isolated instance that can only query and
-	// modify invoices.
-	invoiceMacBytes, err := bakeMacaroon(ctx, svc, invoicePermissions)
+	macaroonBytes, err := bakeMacaroon(ctx, svc, macaroonPermissions)
 	if err != nil {
 		return err
 	}
-	err = ioutil.WriteFile(invoiceFile, invoiceMacBytes, 0644)
+	err = ioutil.WriteFile(filename, macaroonBytes, filePermissions)
 	if err != nil {
-		_ = os.Remove(invoiceFile)
+		_ = os.Remove(filename)
 		return err
 	}
 
-	// Generate the read-only macaroon and write it to a file.
-	roBytes, err := bakeMacaroon(ctx, svc, readPermissions)
-	if err != nil {
-		return err
+	return nil
+}
+
+// genMacaroons checks for three default macaroon files and generates them if
+// they do not exist; one admin-level, one for invoice access and one read-only.
+// Each macaroon is checked and created independently to ensure all three exist.
+// The admin macaroon can also be used to generate more granular macaroons.
+func genMacaroons(ctx context.Context, svc *macaroons.Service,
```

Comment:
nit: rename this to `genDefaultMacaroons` while we're at it? Just to distinguish them from the subserver macaroons that are also created.

Reply:
I preferred this naming too actually but I didn't want to change more than I needed to.